test/packages/knowledge_base/kibana/kb_entry/foo/manifest.yml

```diff
@@ -0,0 +1,14 @@
+name: foo
+description: >
+  Knowledge base entry containing all the foo-related documentation
+index:
+  name: .kibana_kb_foo
+  system: true
+retrieval:
+  syntactic_fields:
+    - content_title
+  semantic_fields:
+    - content_body
+    - ai_subtitle
+    - ai_summary
+    - ai_questions_answered
```

Comment:
Example file of the format specified in `spec/integration/kibana/kb_entry/manifest.spec.yml`.

Most of this manifest is meta that will be used by the KB implementation: 

- `name` and `description` are meta that will be surfaced in the KB management page. `name` will be used at the identifier.
- `retrieval.syntactic_fields` and `retrieval.semantic_fields` are the list of fields that will be used for retrieving documents from this KB entry. Later we could find way to be smarter and more automated about field detection, but I wanted to start with something both easy and concrete, so the KB entry contains those meta. 

- `index.name` is self explanatory - the name of the index that will be created with the specified mappings from the `index-mapping.json` file
- `index.system` is just some future-proofing by exposing the info about the index being Kibana-managed or not (probably don't need more details on this specific PR)

Reply:
> * retrieval.syntactic_fields and retrieval.semantic_fields are the list of fields that will be used for retrieving documents from this KB entry. Later we could find way to be smarter and more automated about field detection, but I wanted to start with something both easy and concrete, so the KB entry contains those meta.

How are these going to be different between different knowledge bases?

> * `index.name` is self explanatory - the name of the index that will be created with the specified mappings from the `index-mapping.json` file

The index name, or anything that could conflict with other resources, should be managed by Fleet. It could be built from the package and entry names.

Reply:
> How are these going to be different between different knowledge bases?

By having those defined in the manifest, and then registered during installation. The search/retrieval code will then forge its request to target the specified fields.

> The index name, or anything that could conflict with other resources, should be managed by Fleet 

I guess that makes sense